build_overrides/asr.gni

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+declare_args() {
+  # ASR IC family.
+  asr_ic_family = ""
+
+  # Root directory for ASR SDK build files.
+  asr_sdk_build_root = ""
+
+  # Root directory for ASR toolchain.
+  asr_toolchain_root = ""
+}
+
+if (asr_ic_family == "") {
+  asr_ic_family = getenv("ASR_IC")
```

Comment:
I think you should not have this environment override at all. You can pass things in via `gn --args` . The fact that the environment only takes effect on the first pass is not intuitive, so I would discourage using getenv. It looks to me like currently only tizen does this and I believe that case is also wrong.

TLDR: dont use getenv in gni files.

Reply:
@andy31415 That's a good idea. But in src/platform/ASR/args.gni ,  I try to use "asr_ic_family" and "asr_sdk_build_root" which already set via --args , but the value is also the same as declare_args{ } defined, do you have any suggestions ?

Reply:
resolved, all asr_ic_family related use --args